samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -262,6 +275,14 @@ public void run(Duration timeout) {
     if (status.getStatusCode() == ApplicationStatus.StatusCode.UnsuccessfulFinish) {
       throw new SamzaException(ExceptionUtils.getStackTrace(status.getThrowable()));
     }
+    // Clean the state directory if any
```

Comment:
We're cleaning the state stores are cleaned up after running the stream application. However,  if there're any exceptions thrown by `runner.run` or `runner.waitForFinish` or if the user stops the test before the completion, this store deletion code flow will not get triggered. So it will be safer to clean the rocksDB state before the run of an application(and we will know that there is no pending state before start).